diff --git a/downloads/lib3mf-2.4.1/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp b/downloads/lib3mf-2.4.1/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp
index b7e39678e..2cc45423f 100644
--- a/downloads/lib3mf-2.4.1/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp
+++ b/downloads/lib3mf-2.4.1/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp
@@ -30,6 +30,8 @@ NMR_ResourcDependencySorter resolves the dependencies between resources by
 sorting them topologically.
 --*/
 
+#include <algorithm>
+
 #include "Model/Writer/v100/NMR_ResourceDependencySorter.h"
 
 #include "Common/Graph/DirectedGraph.h"
@@ -122,4 +124,4 @@ namespace NMR
         return m_resourceIDToIndex.at(resourceID);
     }
 
-}  // namespace NMR
\ No newline at end of file
+}  // namespace NMR
