diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4a7eff..13d2986 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@ include(GenerateExportHeader)
 
 # Set default options
 option(BUILD_EXAMPLE "Build with example" OFF)
+option(BUILD_DEBUG "Build with debug output" OFF)
 option(BUILD_WITH_QT6 "Explicitly build with Qt6" OFF)
 
 # Determine Qt version to use
@@ -27,7 +28,6 @@ set(QtWebDAV_SOURCES
 set(QtWebDAV_HEADERS
     qnaturalsort.h
     qwebdav.h
-    qwebdav_global.h
     qwebdavdirparser.h
     qwebdavitem.h
 )
@@ -60,7 +60,10 @@ target_include_directories(QtWebDAV PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
     $<INSTALL_INTERFACE:include>
 )
-target_compile_definitions(QtWebDAV PRIVATE DEBUG_WEBDAV)
+
+if (BUILD_DEBUG)
+    target_compile_definitions(QtWebDAV PRIVATE DEBUG_WEBDAV)
+endif()
 
 # Link Qt libraries
 target_link_libraries(QtWebDAV PUBLIC
