--- qwt-6.1.4/src/qwt_painter_command.h-old	2021-06-17 14:19:12.111795788 -0400
+++ qwt-6.1.4/src/qwt_painter_command.h	2021-06-17 14:20:12.569318534 -0400
@@ -15,6 +15,7 @@
 #include <qpixmap.h>
 #include <qimage.h>
 #include <qpolygon.h>
+#include <qpainterpath.h>
 
 class QPainterPath;
 
--- qwt-6.1.4/src/qwt_painter.cpp-old	2021-06-17 14:22:58.915761043 -0400
+++ qwt-6.1.4/src/qwt_painter.cpp	2021-06-17 14:23:20.218945853 -0400
@@ -25,6 +25,7 @@
 #include <qabstracttextdocumentlayout.h>
 #include <qstyleoption.h>
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 #include <qapplication.h>
 #include <qdesktopwidget.h>
 
--- qwt-6.1.4/src/qwt_null_paintdevice.cpp-old	2021-06-17 14:24:29.964550953 -0400
+++ qwt-6.1.4/src/qwt_null_paintdevice.cpp	2021-06-17 14:24:44.383676047 -0400
@@ -9,6 +9,7 @@
 
 #include "qwt_null_paintdevice.h"
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 #include <qpixmap.h>
 
 class QwtNullPaintDevice::PrivateData
--- qwt-6.1.4/src/qwt_widget_overlay.cpp-old	2021-06-17 14:28:22.809571023 -0400
+++ qwt-6.1.4/src/qwt_widget_overlay.cpp	2021-06-17 14:28:32.922658759 -0400
@@ -10,6 +10,7 @@
 #include "qwt_widget_overlay.h"
 #include "qwt_painter.h"
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qpaintengine.h>
 #include <qimage.h>
 #include <qevent.h>
--- qwt-6.1.4/src/qwt_plot_renderer.cpp-old	2021-06-17 14:29:56.625384927 -0400
+++ qwt-6.1.4/src/qwt_plot_renderer.cpp	2021-06-17 14:30:28.524661671 -0400
@@ -20,6 +20,7 @@
 #include "qwt_math.h"
 
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qtransform.h>
 #include <qprinter.h>
 #include <qfiledialog.h>
--- qwt-6.1.4/src/qwt_plot_panner.cpp-old	2021-06-17 14:32:20.297631352 -0400
+++ qwt-6.1.4/src/qwt_plot_panner.cpp	2021-06-17 14:32:47.424866689 -0400
@@ -14,6 +14,7 @@
 #include <qbitmap.h>
 #include <qstyle.h>
 #include <qstyleoption.h>
+#include <qpainterpath.h>
 
 #if QT_VERSION >= 0x050000
 #if QT_VERSION < 0x050100
--- qwt-6.1.4/src/qwt_painter.h-old	2021-06-17 14:34:21.721684771 -0400
+++ qwt-6.1.4/src/qwt_painter.h	2021-06-17 14:34:37.623822732 -0400
@@ -17,6 +17,7 @@
 #include <qpen.h>
 #include <qline.h>
 #include <qpalette.h>
+#include <qpainterpath.h>
 
 class QPainter;
 class QBrush;
--- qwt-6.1.4/src/qwt_plot_glcanvas.h-old	2021-06-17 14:37:26.537288151 -0400
+++ qwt-6.1.4/src/qwt_plot_glcanvas.h	2021-06-17 14:37:42.282424753 -0400
@@ -11,6 +11,7 @@
 #define QWT_PLOT_GLCANVAS_H
 
 #include "qwt_global.h"
+#include <qpainterpath.h>
 #include <qframe.h>
 #include <qgl.h>
 
