diff --git a/TheForceEngine/TFE_FileSystem/paths-posix.cpp b/TheForceEngine/TFE_FileSystem/paths-posix.cpp
index 02a79f8d..38c9eea5 100644
--- a/TheForceEngine/TFE_FileSystem/paths-posix.cpp
+++ b/TheForceEngine/TFE_FileSystem/paths-posix.cpp
@@ -98,9 +98,7 @@ namespace TFE_Paths
 		}

 		std::string s;
-		s = std::string("/usr/local/share/") + append + "/";
-		s_systemPaths.push_back(s);
-		s = std::string("/usr/share/") + append + "/";
+		s = std::string("/app/share/") + append + "/";
 		s_systemPaths.push_back(s);
 		setTFEPath(append, PATH_PROGRAM_DATA);
 		s_systemPaths.push_back(getPath(PATH_PROGRAM_DATA));
@@ -119,7 +117,7 @@ namespace TFE_Paths
 		// ensure SetProgramDataPath() was called before
 		assert(!s_paths[PATH_PROGRAM_DATA].empty());

-		s_paths[PATH_USER_DOCUMENTS] = s_paths[PATH_PROGRAM_DATA];
+		s_paths[PATH_USER_DOCUMENTS] = strcat(getenv("XDG_CONFIG_HOME"), "/");
 		s_systemPaths.push_back(getPath(PATH_USER_DOCUMENTS));
 		// set cwd to user documents path, because Dear ImGUI drops
 		// its ini file at the cwd, we don't want it to litter somewhere
